manager/allocator/network.go

```diff
@@ -641,6 +644,34 @@ func (a *Allocator) commitAllocatedNode(ctx context.Context, batch *store.Batch,
 	return nil
 }
 
+func updatePortsInHostPublishMode(s *api.Service) {
+	if s.Endpoint != nil {
+		var portConfigs []*api.PortConfig
+		for _, portConfig := range s.Endpoint.Ports {
+			if portConfig.PublishMode == api.PublishModeIngress {
+				portConfigs = append(portConfigs, portConfig)
+			}
+		}
+		s.Endpoint.Ports = portConfigs
+	}
+
+	if s.Spec.Endpoint != nil {
+		if s.Endpoint == nil {
+			s.Endpoint = &api.Endpoint{}
+		}
+		if s.Endpoint.Spec == nil {
+			s.Endpoint.Spec = &api.EndpointSpec{}
+		}
+		s.Endpoint.Spec.Ports = s.Spec.Endpoint.Ports
```

Comment:
I believe this should be `s.Endpoint.Spec.Ports = s.Spec.Endpoint.Ports.Copy()`. We don't want these slices to be backed by the same memory.